### PR TITLE
Circuit extensions

### LINKED ISSUE #11: Neural Circuit extensions
There are some additional features we should be adding (or at least testing) for our networks:
- [x] Batch/Layer/InstanceNormalization. Initial test seemed to indicate InstanceNormalization can improve performance.
- [x] Make the target of reconstruction a parameter of the reconstruction loss (rather than forcing it to be "decoder").
- [x] Torchinfo should scan the whole brain.

### PR DESCRIPTION
**Pull request by alex404**
Resolves #11 

List of changes:

- Added gelus
- Moved circuit construction out of brain, so that `brain.py` doesn't have to depend on Hydra anymore.
- Added a global brain scan with torchinfo. This complements the circuit scans nicely.
- use_wandb=false by default
- Fixed overly complex `runner/analysis.py` core function.
- Generalized reconstruction error loss to allow the decoder name to be selected.
- added instance layer normalization which can be toggled.

Still wrestling with the fact that in certain regimes the receptive fields across all layers exhibit weird artifacts (basically, it looks like there's a green/red 45 degree gabor overlayed on all RFs). This seems to have emerged though because I changed the defaults of the image transformations to be less extreme. When I switched back to the older config this "fixed" the artifacts. It seems like it might be some kind of overfitting thing, but weight_decay doesn't help. I'll keep running tests. This seems like less of a bug or more like a training issue at this point, so I think we're ready to merge.

### CODE REVIEW

File: runner/analyze.py
Comment (fabioseel) [inline runner/analyze.py:1]: In general I won't bother doing an extensive review, do you think you can decrease the logical complexity of the analyze method ruff complains about?
Else we can merge imho

**Comment by alex404** [comment]
So I think I resolved everything now (e.g. the ruff complexity complaint), but one issue is that I think ruff changed it's import sorting algorithm between 0.5.4 (what's in the image) and the latest version that I use locally (0.7.0). I changed ruff to 0.7.0 in the def file, and that seemed to build fine, and then I passed the checks. However, the auto rebuild mechanism in the CI seems to have failed. Can you double check this @fabioseel?

**Comment by fabioseel** [comment]
For me, also when running ruff 0.7.0 locally reports the same import unsorted/unformatted error.
Is it possible you have some weird config set or lying around somewhere?
